backend/src/main/java/de/sambam/gamenighttracker/controller/UserController.java

```diff
@@ -22,32 +24,40 @@ public UserController(UserService userService) {
     }
 
     @GetMapping("/games")
-    public List<Game> getGameList() {
-        return userService.listAllGames("1");
+    public List<Game> getGameList(Principal principal) {
+        String loggedInUsername = principal.getName();
+        return userService.listAllGames(loggedInUsername);
     }
 
     @GetMapping("/players")
-    public List<PlayerDto> getPlayerList() {
-        return userService.listAllPlayers("1");
+    public List<PlayerDto> getPlayerList(Principal principal) {
+        String loggedInUsername = principal.getName();
+        return userService.listAllPlayers(loggedInUsername);
     }
 
     @PostMapping("/game")
-    public Optional<Game> addNewGame(@RequestBody Game gameToAdd) {
+    public Optional<Game> addNewGame(@RequestBody Game gameToAdd, Principal principal) {
+        String loggedInUsername = principal.getName();
         try {
-            return userService.addNewGame(gameToAdd, "1");
+            return userService.addNewGame(gameToAdd, loggedInUsername);
         } catch (Exception e) {
             throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "No gamelist was generated!");
         }
```

Comment:
Is there a reason why you `try {} catch () {}` instead of return an `Optional.empty()` from the userService in case of missing gameList? Moreover a Http Error with Status Code 500 feels like something went totaly wrong. Perhaps a more differentiated error throwing would be more user friendly (u could handle this in the frontend even better...)

Reply:
not sure if my approach is correct, but the error should only occur, when for some reason the nesting object is not generated in the backend. and the message would point me to the respective problem... at this point I'm only considering me as a user ^^'

Reply:
You may remember our conversation regarding expandability :)